### ISSUE TITLE
Stabilise Compose Animations API

### ISSUE THREAD

**Issue by arkivanov** [arkivanov/Decompose#110]
Docs: [link](https://arkivanov.github.io/Decompose/extensions/compose/#animations-experimental). Collecting feedback and use cases here.

Points to improve:

- [x] Get rid of `Direction.IDLE` using `movableContentOf` (blocked by Compose 1.2.0 release)
- [x] Make click interception configurable? Or replace with another solution? See [b/214231672](https://issuetracker.google.com/issues/214231672)
- [x] Nullable `StackAnimator` in `StackAnimation` selector
- [x] Shared element transitions (perhaps `LookaheadLayout` + `movableContentOf` should work out of the box when they land in Compose)
- [x] Add `otherChild` argument to `stackAnimation` selector
- [x] Add `minAlpha` argument to `fade` animation

**Comment by plusmobileapps** [comment]
It would be nice to have a shared element transition between configurations, not sure if something similar to this [sample](https://github.com/mobnetic/compose-shared-element) could be done with Decompose?

**Comment by arkivanov** [comment]
I have a pending branch that should make it compatible with the official shared transitions (`LookaheadLayout` + `movableContentOf`) when they land in Compose. The branch is blocked by the availability of `movableContentOf`. Waiting for the stable MPP Compose 1.2.0.

**Comment by arkivanov** [comment]
Actually, I found that `LookaheadLayout` is already available to try in Compose `1.3.0-alpha01`. I will play with it and provide some updates here.

**Comment by arkivanov** [comment]
Yep, `LookaheadLayout` + `movableContentOf` should work just fine even with the current animations API. It is already available in Compose `1.3.0-alpha01`, however there are immediate bugs, e.g. [LookaheadLayout doesn't support having LazyColumn inside](https://issuetracker.google.com/issues/240016022), etc.

https://user-images.githubusercontent.com/26204457/180651812-462d4661-20e3-48e8-a0cb-6d341843e3c2.mov

**Comment by plusmobileapps** [comment]
It seems like the [animation documentation](https://arkivanov.github.io/Decompose/extensions/compose/#animations-experimental) is out of date since the stack navigation changes. I'm assuming `childAnimation` is supposed to be `stackAnimation`?

**Comment by arkivanov** [comment]
Thanks for raising! I updated the documentation a while ago with multiple rounds of self code review, but seems like I missed this one.

**Comment by aartikov** [comment]
@arkivanov Hi. Is it possible to use [AnimatedContent](https://developer.android.com/jetpack/compose/animation#animatedcontent) with Decompose?

**Comment by arkivanov** [comment]
@aartikov Technically it should be possible. E.g. the `Child Stack` just exposes `Value<ChildStack<C, T>>` - so you can use something your own instead of the `Children` function, with `AnimatedContent`. You can also try the [StackAnimation { }](https://arkivanov.github.io/Decompose/extensions/compose/#custom-animations) function that just provides the current `ChildStack` and it's up to you how to render children.

**Comment by aartikov** [comment]
@arkivanov Thank you!
